eng/scripts/Prepare-Release.ps1

```diff
@@ -0,0 +1,125 @@
+param($package)
+
+$repoRoot = Resolve-Path "$PSScriptRoot/../..";
+
+. ${repoRoot}\eng\common\scripts\SemVer.ps1
+. ${repoRoot}\eng\common\scripts\ChangeLog-Operations.ps1
+
+$packageDirectory = Get-ChildItem "$repoRoot/sdk" -Directory -Recurse -Depth 2 -Filter $package
+$serviceDirectory = ($packageDirectory).Parent.Name
+
+Write-Host "Source directory $serviceDirectory"
+
+$existing = Invoke-WebRequest "https://api.nuget.org/v3-flatcontainer/$($package.ToLower())/index.json" | ConvertFrom-Json;
+
+$libraryType = "preview";
+$latestVersion = "unknown";
+foreach ($existingVersion in $existing.versions)
+{
+    $parsedVersion = [AzureEngSemanticVersion]::new($existingVersion)
+    if (!$packageOldSemVer.IsPrerelease)
+    {
+        $libraryType = "GA"
+    }
+    $latestVersion = $existingVersion;
+}
+
+$latestVersion = $existing.versions[$existing.versions.Count - 1];
+
+Write-Host
+Write-Host "Latest released version $latestVersion, library type $libraryType" -ForegroundColor Green
+
+$newVersion = Read-Host -Prompt 'Input the new version' 
+
+$releaseType = "None";
+$parsedNewVersion = [AzureEngSemanticVersion]::new($newVersion)
+if ($parsedNewVersion.Major -ne $parsedVersion.Major)
+{
+    $releaseType = "Major"
+}
+elseif ($parsedNewVersion.Minor -ne $parsedVersion.Minor)
+{
+    $releaseType = "Minor"
+}
+elseif ($parsedNewVersion.Patch -ne $parsedVersion.Patch)
+{
+    $releaseType = "Bugfix"
+}
+elseif ($parsedNewVersion.IsPrerelease)
+{
+    $releaseType = "Bugfix"
+}
+
+Write-Host
+Write-Host "Detected released type $releaseType" -ForegroundColor Green
+
+Write-Host
+Write-Host "Updating versions" -ForegroundColor Green
+
+& "$repoRoot\eng\scripts\Update-PkgVersion.ps1" -ServiceDirectory $serviceDirectory -PackageName $package -NewVersionString $newVersion
+
+$date = Get-Date
+$month = $date.ToString("MMMM")
+if ($date.Day -gt 15)
+{
+    $month = $date.AddMonths(1).ToString("MMMM")
+}
+Write-Host
+Write-Host "Assuming release is in $month" -ForegroundColor Green
+
+$commonParameter = @("--organization", "https://dev.azure.com/azure-sdk", "-o", "json", "--only-show-errors")
+
+$workItems = az boards query @commonParameter --project Release --wiql "SELECT [ID], [State], [Iteration Path], [Title] FROM WorkItems WHERE [State] <> 'Closed' AND [Iteration Path] under 'Release\2020\$month' AND [Title] contains '.NET'" | ConvertFrom-Json;
```

Comment:
If this command doesn't exist, the script will move on by default. Best set `$ErrorPreference = 'Stop'` at the top.

Reply:
In fact, since you're making incremental changes, you might even assert it exists earlier like so:

```powershell
if (!(get-command az)) {
  throw 'You must have the Azure CLI installed: https://aka.ms/azure-cli'
}
```